common/client.go

```diff
@@ -22,10 +41,26 @@ type DatabricksClient struct {
 
 	// callback used to create API1.2 call wrapper, which simplifies unit tessting
 	commandFactory func(context.Context, *DatabricksClient) CommandExecutor
+	cachedUser     scim.User
 }
 
 func (c *DatabricksClient) WorkspaceClient() (*databricks.WorkspaceClient, error) {
-	return databricks.NewWorkspaceClient((*databricks.Config)(c.DatabricksClient.Config))
+	// return databricks.NewWorkspaceClient((*databricks.Config)(c.DatabricksClient.Config))
+	w, err := databricks.NewWorkspaceClient((*databricks.Config)(c.DatabricksClient.Config))
```

Comment:
if we're creating a workspace client every time we do `c.WorkspaceClient()`, there's no caching ;)

Reply:
add mutex here

Reply:
This should be fine because the address of cachedUser is not going to change.